### ISSUE TITLE
[CRASH] Decode gif image to static image, and it could lead to JNI Crash.

### ISSUE THREAD

**Issue by huangjunkun** [facebook/fresco#1350]
**Decode gif image to static image, and it could lead to JNI Crash**

> com.facebook.imagepipeline.decoder.ImageDecoder#decodeGif
> 
> com.facebook.imagepipeline.decoder.ImageDecoder#decodeStaticImage

**Crash on Android 4.4.4, 4.1.2, 4.3, 4.4.2**

**JNI Crash log as follow** 

```
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr deadbaad
  r0 00000000  r1 40136456  r2 deadbaad  r3 40139fbb
  r4 7cdac0b8  r5 40144180  r6 400e5000  r7 7cdac0c0
  r8 7c5f77d0  r9 7c5f73a8  10 7c5f7378  fp 00000200
  ip 00000001  sp 7c5f72d0  lr 4010771f  pc 40107720  cpsr 600f0030
    --- --- ---
    00 pc 00011720  /system/lib/libc.so (dlfree+1191)
    01 pc 0000dc6f  /system/lib/libc.so (free+10)
    02 pc 000fe179  /system/lib/libskia.so (_ZN17SkGIFImageDecoder8onDecodeEP8SkStreamP8SkBitmapN14SkImageDecoder4ModeE+1064)
    03 pc 000fc41d  /system/lib/libskia.so (_ZN14SkImageDecoder6decodeEP8SkStreamP8SkBitmapNS2_6ConfigENS_4ModeE+36)
    04 pc 001013ab  /system/lib/libskia.so (_ZN21SkImageRef_GlobalPool8onDecodeEP14SkImageDecoderP18SkStreamRewindableP8SkBitmapNS4_6ConfigENS0_4ModeE+12)
    05 pc 00100e17  /system/lib/libskia.so (_ZN10SkImageRef13prepareBitmapEN14SkImageDecoder4ModeE+102)
    06 pc 00100e5d  /system/lib/libskia.so (_ZN10SkImageRef12onLockPixelsEPP12SkColorTable+12)
    07 pc 000c969b  /system/lib/libskia.so (_ZN10SkPixelRef10lockPixelsEv+38)
    08 pc 0009bc77  /system/lib/libskia.so (_ZNK8SkBitmap10lockPixelsEv+20)
    09 pc 00100cd7  /system/lib/libskia.so (_ZN10SkImageRef8isOpaqueEP8SkBitmap+18)
    10 pc 00077135  /system/lib/libandroid_runtime.so
    11 pc 0007732b  /system/lib/libandroid_runtime.so
    12 pc 0001d30c  /system/lib/libdvm.so (dvmPlatformInvoke+112)
    13 pc 0004d8db  /system/lib/libdvm.so (_Z16dvmCallJNIMethodPKjP6JValuePK6MethodP6Thread+398)
    14 pc 00026720  /system/lib/libdvm.so
    15 pc 0002d790  /system/lib/libdvm.so (_Z11dvmMterpStdP6Thread+76)
    16 pc 0002adf4  /system/lib/libdvm.so (_Z12dvmInterpretP6ThreadPK6MethodP6JValue+184)
    17 pc 0005fd75  /system/lib/libdvm.so (_Z14dvmCallMethodVP6ThreadPK6MethodP6ObjectbP6JValueSt9__va_list+336)
    18 pc 0005fd99  /system/lib/libdvm.so (_Z13dvmCallMethodP6ThreadPK6MethodP6ObjectP6JValuez+20)
    19 pc 00054a93  /system/lib/libdvm.so
    20 pc 0000d1d0  /system/lib/libc.so (__thread_entry+72)
    21 pc 0000d368  /system/lib/libc.so (pthread_create+240)
    --- --- --- ---
```

**Java backtrace**

```
at android.graphics.BitmapFactory.nativeDecodeByteArray
  at android.graphics.BitmapFactory.decodeByteArray
  at com.facebook.imagepipeline.platform.KitKatPurgeableDecoder.decodeByteArrayAsPurgeable
  at com.facebook.imagepipeline.platform.DalvikPurgeableDecoder.decodeFromEncodedImage
  at com.facebook.imagepipeline.decoder.ImageDecoder.decodeStaticImage
  at com.facebook.imagepipeline.decoder.ImageDecoder.decodeGif
  at com.facebook.imagepipeline.decoder.ImageDecoder.decodeImage
  at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.doDecode
  at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.onNewResultImpl

```

**Comment by aagnes** [comment]
Thanks for reporting the bug, what Android device are you using? It seems you're testing in 4.4.4, 4.1.2, 4.3, 4.4.2, have you encountered the native crash in >=5.0 devices? And also which Fresco library version are you using?

**Comment by huangjunkun** [comment]
@aagnes    
Yeah, Our Product using Fresco's ImagePipeline has many user. The crash occur on Android  4.4.4, 4.1.2, 4.3, 4.4.2, not Found on Android 5.0 at present.

**Comment by oprisnik** [comment]
We've released [Fresco v1.4.0](https://github.com/facebook/fresco/releases/tag/v1.4.0), which includes a new implementation for animated images, which might have fixed this issue. Let us know if you're still experiencing problems with 1.4.0 and we can re-open it.
